### PR TITLE
feat: Google Gemini

### PR DESCRIPTION
**Pull request by gunpal5**
Added Google Gemini Basic Support.

https://github.com/tryAGI/LangChain/issues/89

### CODE REVIEW

File: src/libs/Providers/LangChain.Providers.Google/LangChain.Providers.Google.csproj [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    <ItemGroup>
+      <PackageReference Include="Google_GenerativeAI" />
+    </ItemGroup>
+
+    <ItemGroup Label="Usings">
+        <Using Include="System.Net.Http" />
+    </ItemGroup>
+
+    <PropertyGroup Label="NuGet">
+        <Description>HuggingFace API LLM and Chat model provider.</Description>
```
Comment (HavenDV) [inline src/libs/Providers/LangChain.Providers.Google/LangChain.Providers.Google.csproj:17]: Please change description and tags

File: src/tests/LangChain.Providers.Google.UnitTests/GenerativeModelTests.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
+using LangChain.Providers.Models;
+
+namespace LangChain.Providers.OpenAI.UnitTests;
+
+[TestFixture]
```
Comment (HavenDV) [inline src/tests/LangChain.Providers.Google.UnitTests/GenerativeModelTests.cs:5]: Mark with [Explicit] attribute

File: src/libs/Providers/LangChain.Providers.Google/LangChain.Providers.Google.csproj (hunk 1 above)
Comment (gunpal5) [reply src/libs/Providers/LangChain.Providers.Google/LangChain.Providers.Google.csproj]: Sorry about that, I changed the description

File: src/tests/LangChain.Providers.Google.UnitTests/GenerativeModelTests.cs (hunk 2 above)
Comment (gunpal5) [reply src/tests/LangChain.Providers.Google.UnitTests/GenerativeModelTests.cs]: Added Explicit Attribute

**Comment by HavenDV** [comment]
Are you planning to add anything else or it could be merged?

**Comment by gunpal5** [comment]
@HavenDV I was thinking to add function calling as well. Its already coded in base library, but not sure how to implement it for LangChain, it might conflict with the OpenAI generators.

**Comment by HavenDV** [comment]
@all-contributors please add @gunpal5 for infrastructure, tests and code

**Pull request merged**